Sources/Common/Communication/EventBusHandler.swift

```diff
@@ -0,0 +1,113 @@
+import Combine
+import Foundation
+
+/// `EventBusHandler` acts as a central hub for managing events in the application.
+/// It interfaces with both an event bus for real-time event handling and an event storage system for persisting events.
+// sourcery: InjectRegisterShared = "EventBusHandler"
+// sourcery: InjectSingleton
+public class EventBusHandler {
+    private let eventBus: EventBus
+    private let eventStorage: EventStorage
+    private let memoryStorage: MemoryStorage = .init()
+    private let logger: Logger
+
+    /// Initializes the EventBusHandler with dependencies for event bus and storage.
+    /// - Parameters:
+    ///   - eventBus: An instance of EventBus to handle event posting and observer management.
+    ///   - eventStorage: An instance of EventStorage to manage event persistence.
+    ///   - logger: A logger for logging information and errors.
+    /// Automatically loads events from file-based storage into in-memory storage upon initialization.
+    public init(eventBus: EventBus, eventStorage: EventStorage, logger: Logger) {
+        self.eventBus = eventBus
+        self.eventStorage = eventStorage
+        self.logger = logger
+        Task { await loadEventsFromStorage() }
+    }
+
+    /// Loads events from persistent storage into in-memory storage for quick access and event replay.
+    private func loadEventsFromStorage() async {
+        for eventType in EventTypesRegistry.allEventTypes() {
+            do {
+                let key = eventType.key
+                let events: [AnyEventRepresentable] = try await eventStorage.loadEvents(ofType: key)
+                await memoryStorage.storeEvents(events, forKey: key)
+            } catch {
+                logger.debug("Error loading events for \(eventType): \(error)")
+            }
+        }
+    }
+
+    /// Adds an observer for a specific event type and replays any stored events of that type to the new observer.
+    /// - Parameters:
+    ///   - eventType: The event type to observe.
+    ///   - action: The action to execute when the event is observed.
+    public func addObserver<E: EventRepresentable>(_ eventType: E.Type, action: @escaping (E) -> Void) {
+        logger.debug("EventBusHandler: Adding observer for event type - \(eventType)")
+
+        let adaptedAction: (AnyEventRepresentable) -> Void = { event in
+            if let specificEvent = event as? E {
+                action(specificEvent)
+            } else {
+                self.logger.debug("Error: Event type did not match")
+            }
+        }
+
+        Task {
+            await eventBus.addObserver(eventType.key, action: adaptedAction)
+            await replayEvents(forType: eventType)
+        }
```

Comment:
Doing so, `addObserver` remains synchronous from the caller's perspective. The async work is managed internally without requiring the caller to handle it or even be aware of it. Makes the public API easier to adapt, every module subscribing won't have to add `Task` on their end. 

Since registering an observer is a setup process that doesn't need to be tightly synchronized with other operations, using a Task made sense. For instance, our modules simply need to start listening for events and don’t have subsequent actions that depend on the successful registration hence a fire-and-forget approach. 

It also ensures consistency of experience for all modules subscribing. Similar to what NotificationCenter API offers.